lib/Service.php

```diff
@@ -230,6 +230,33 @@ function mapValueObject($elementName, $className) {
         $this->classMap[$className] = function(Writer $writer, $valueObject) use ($namespace) {
             return \Sabre\Xml\Serializer\valueObject($writer, $valueObject, $namespace);
         };
+        $this->valueObjectMap[$className] = $elementName;
```

Comment:
The use-case is if an application provides a service, and a module alters the standard behavior the application provides. I don't think that that's entirely far-fetched.